src/Pkg.jl

```diff
@@ -39,31 +28,310 @@ include("Operations.jl")
+
+Run the build script in deps/build.jl for `pkg` and all of the dependencies in
+depth-first recursive order.
+If no argument is given to `build`, the current project is built, which thus needs
+to be a package.
+This function is called automatically one any package that gets installed
+for the first time.
+
+"""
+const build = API.build
+
+# TODO: decide what to do with this
+const installed = API.installed
+
+"""
+    Pkg.pin(pkg::Union{String, Vector{String})
+    Pkg.pin(pkgs::Union{Packagespec, Vector{Packagespec}})
+
+Pin a package to the current version (or the one given in the `packagespec`.
+a pinned package will never be updated.
+"""
+const pin = API.pin
+
+"""
+    Pkg.free(pkg::Union{String, Vector{String})
+    Pkg.free(pkgs::Union{Packagespec, Vector{Packagespec}})
+
+Free a package which removes a `pin` if it exists, or if the package is tracking a path,
+e.g. after [`Pkg.develop`](@ref), go back to tracking registered versions.
+
+## Examples
+
+```
+Pkg.free("Package")
+Pkg.free(PackageSpec("Package"))
+```
+"""
+const free = API.free
+
+
+"""
+    Pkg.develop(pkg::Union{String, Vector{String})
+    Pkg.develop(pkgs::Union{Packagespec, Vector{Packagespec}})
+
+Makes a package available for development by tracking it by path.
+If `pkg` is given with only a name or by a URL the packages will be downloaded
+to the location by the environment variable `JULIA_PKG_DEVDIR` with
+`.julia/dev` as the default.
+
+If `pkg` is given as a local path, the package at that path will be tracked.
+
+## Examples
+
+```
+# By name
+Pkg.develop("Example")
+
+# By url
+Pkg.develop(PackageSpec(url="https://github.com/JuliaLang/Compat.jl", rev="master"))
+
+# By path (also uses url keyword to PackageSpec)
+Pkg.develop(PackageSpec(url="MyJuliaPackages/Package.jl")
+```
+
+See also [`PackageSpec`](@ref)
+
+"""
+const develop = API.develop
+
+#TODO: Will probably be deprecated for something in PkgDev
+const generate = API.generate
+
+"""
+    Pkg.instantiate()
+
+If a `Manifest.toml` file exist in the current project, download all
+the packages declared in that manifest.
+Else, resolve a set of feasible packages from the `Project.toml` files
+and install them.
+"""
+const instantiate = API.instantiate
+
+"""
+    Pkg.resolve()
+
+Update the current manifest with eventual changes to the dependency graph
+from packages that are tracking a path.
+"""
+const resolve = API.resolve
+
+"""
+    Pkg.status(mode::PackageMode=PKGMODE_PROJECT)
+
+Prints out the status of the project/manifest.
+If `mode` is `PKGMODE_PROJECT` prints out status about only those packages
+that are in the project (explicitly added). If `mode` is `PKGMODE_MANIFEST`
+also print for those in the manifest (recursive dependencies).
+"""
+const status = API.status
+
+
+"""
+    Pkg.activate([s::String])
+
+Activate the environment at `s`. The active environment is the environment
+that the package manager modifies when it runs command.
```

Comment:
The active environment is the environment that is modified by executing package commands.